File: backend/schemas/dialog.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from schemas.user import UserResponse

class MessageBase(BaseModel):
    content: str

class MessageCreate(MessageBase):
    dialog_id: int

class MessageResponse(MessageBase):
    id: int
    sender_id: int
    dialog_id: int
    is_read: bool
    created_at: datetime
    sender: UserResponse
    
    class Config:
        from_attributes = True

class DialogResponse(BaseModel):
    id: int
    user1_id: int
    user2_id: int
    created_at: datetime
    updated_at: datetime
    other_user: UserResponse
    last_message: Optional[MessageResponse] = None
    unread_count: int = 0
    
    class Config:
        from_attributes = True

class DialogCreate(BaseModel):
    user_id: int

File: frontend/src/types/dialog.ts
import type { UserResponse } from "./auth";

export interface DialogResponse {
  id: number;
  user1_id: number;
  user2_id: number;
  created_at: string;
  updated_at: string;
  other_user: UserResponse;
  last_message?: MessageResponse;
  unread_count: number;
}

export interface MessageResponse {
  id: number;
  sender_id: number;
  dialog_id: number;
  content: string;
  is_read: boolean;
  created_at: string;
  sender: UserResponse;
}